tools/roxvet/analyzers/dontcompareproto/analyzer.go

```diff
@@ -0,0 +1,124 @@
+package donotcompareproto
+
+import (
+	"fmt"
+	"go/ast"
+	"go/types"
+	"strings"
+
+	"github.com/stackrox/rox/pkg/set"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+	"golang.org/x/tools/go/types/typeutil"
+)
+
+const doc = `Inspect calls to Equal for proto arguments that should be compared with protocompat.Equal instead`
+
+// Analyzer is the go vet entrypoint
+var Analyzer = &analysis.Analyzer{
+	Name:     "donotcompareproto",
+	Doc:      doc,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+	Run:      run,
+}
+
+var (
+	protoPkgs = []string{
+		"*github.com/stackrox/scanner/generated",
+		"*github.com/stackrox/rox/generated/",
+	}
+
+	replacements = map[string]string{
+		"":              "Equal",
+		"[]":            "SlicesEqual",
+		"map[string][]": "MapSliceEqual",
+		"map[string]":   "MapEqual",
+	}
+
+	allowedCallerPackages = []string{
+		"github.com/stackrox/rox",
+	}
+
+	bannedEqualFunctions = set.NewFrozenStringSet(
+	//"github.com/google/go-cmp/cmp.Equal",
+	//"reflect.DeepEqual",
```

Comment:
Since we're apparently excluding all caller paths at the moment, would it make sense to uncomment the banned functions ?